src/sentry/tasks/post_process.py

```diff
@@ -626,82 +625,36 @@ def get_event_raise_exception() -> Event:
                     event=event,
                 )
 
-        # TODO: Remove this check once we're sending all group ids as `group_states` and treat all
-        # events the same way
-        if not is_transaction_event and group_states is None:
```

Comment:
@wedamija aren't there cases where we pass in`group_state` with length 1?

Reply:
Yep, whenever `group_state` is passed it'll have a length of one. The point though is we can always just construct that from the `group_id` and related params, so we don't need to use `group_states`